### ISSUE TITLE
Allow author to mention non-participants in comments to a private post

### ISSUE THREAD

**Issue by cmrd-senya** [diaspora/diaspora#7390]
Originally raised here: https://github.com/diaspora/diaspora/issues/1851#issuecomment-288593798.

Now it is implemented so that in a comment to a private post we can mention only participants (commenters and likers). But I see another usecase: what if you are a private post author, and you forgot to mention someone in the post, or you decided to call some to join the discussion. Why can't you mention your contact? Since you're the post author, we have all the information to send a mention in comment notification to some of your contacts. I mostly talk about the case, when your post has that person in original recipients.

So I propose to add a rule: if you're a private post author, you're allowed to mention in a comment any member of target aspects of this post.

However we can't implement it right away, @SuperTux88 says:

> because we still have the problem, that we don't know who can see the post, only which aspects were selected when creating it (so right after posting it is the same, but not after you changed some aspect-memberships). So I think we should first fix that, and then can discuss again if we want to enable it for post-authors. https://github.com/diaspora/diaspora/issues/1851#issuecomment-288898112

There is another scenario, that can be considered: it could be also possible to allow the author to mention this way people, who aren't members of the target aspects. This would imply that when the mentioning comment is posted, the post and the comment is pushed to the mentioned person.

The opinions stated on this case so far:

@goobertron:
> There was a conscious decision not to allow a widening of the scope of readership of a post after it has been created, for various reasons, including so that people who have commented on a post in the understanding that it is limited to a certain readership don't later find that a wider readership can now see their comments. This proposal would enable this, albeit on a small scale (individually mentioned people). I'm not sure the usefulness of this feature would outweigh possible negative consequences. https://github.com/diaspora/diaspora/issues/1851#issuecomment-288617918

@SuperTux88:
> This is a completely different topic and should have it's own discussion. Personally I think it's ok (as long as the post stays private, so no change privayte -> public). Because the receivers of the post only see that it is private, not how many people can see it (could be only you and the author, could be "all aspects" of the author). So it doesn't make a difference as reader, if the author adds a person after they posted it.
>
> But again, out of scope for mentions and probably also only possible after fixing the problem above, because we can't track single-person targets currently https://github.com/diaspora/diaspora/issues/1851#issuecomment-288898112

So to sum up, even if we find this desirable we need to fix the "single person targets issue". The second proposal in this topic also requires some more discussion.

**Comment by cmrd-senya** [comment]
I'm not quite sure the "ui" label fits here, because the issue is mainly related to the backend.

**Comment by SuperTux88** [comment]
I removed `ui` and added `feature`, I hope that fits better.
